components/tools/OmeroWeb/test/integration/test_histogram.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+# Copyright (C) 2016 University of Dundee & Open Microscopy Environment.
+# All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests rendering of thumbnails."""
+
+from omeroweb.testlib import IWebTest
+from omeroweb.testlib import _get_response_json
+
+import pytest
+from django.core.urlresolvers import reverse
+
+
+class TestHistogram(IWebTest):
+    """Tests loading of histogram json data."""
+
+    @pytest.mark.parametrize("bins", [None, 10])
+    def test_histogram_bin_count(self, bins):
+        """
+        Test that the we get histogram json of the expected size.
+
+        Default size is 256 bins.
+        """
+        size_x = 125
+        size_y = 125
+        img_id = self.create_test_image(size_x=size_x, size_y=size_y,
+                                        session=self.sf).id.val
```

Comment:
understood so will need to go over the inconsistency I have noticed in various locations
The style probably evolves through time